Enterprise-CMCS/macpro-mdct-mcr
# Fix unmarshaller and add script to re-run kafka sync for malformed records

### Description


The unmarshaller is part of the kafka sync process, which runs when we update a DB or S3 record. Thus the easiest way to fix the malformed records is to correct the unmarshaller and re-fire all events that could have happened since the issue started (earlier this year). This PR does the following:

To fix the issue:
- Fix unmarshaller by removing options

To reprocess malformed records:
- Add trigger to kafka lambda on S3 Object Tagging
  - This allows us to re-submit the records by adding a tag, as opposed to having to touch the record and possibly interfering with state users
- Add sync-kafka-2024 script to be run against each environment to re-trigger record processing for 2024
  - Add dynamodb and s3 utilities to support our script
  - Add aws-sdk-v3 packages for utilities

References:
[ListObjectsV2Command](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/s3/command/ListObjectsV2Command/)
[PutObjectTaggingCommand](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/s3/command/PutObjectTaggingCommand/)

[original fix in CARTS](https://github.com/Enterprise-CMCS/macpro-mdct-carts/pull/139639)

### Related ticket(s)


CMDCT-3695

---
### How to test

Prep:
- In services/uploads/ delete your `local_buckets/` folder to clear out past local reports
- Run `./run local`
- In a new terminal tab run `DYNAMO_ENDPOINT=http://localhost:8000 dynamodb-admin`
- In a new terminal tab in MCR root run `DYNAMODB_URL="http://localhost:8000" S3_LOCAL_ENDPOINT="http://localhost:4569" node services/database/scripts/sync-kafka-2024.js`
- See in the console that the output says there were no tables and no buckets touched
  - You can search the terminal output for "Touched". There should be six logs: three for db and three for buckets
- In a browser go to localhost:8001 to see your local dynamo instance
- In another browser tab go to localhost:3000 to see MCR
- Create a MCPAR and MLR report
- In your services/uploads/local_buckets folder view the MCPAR and MLR buckets. See that in the `fieldData/` folder there are three files all starting with the same id (these all pertain to the single report you created)
- At localhost:8001 view your MCPAR and MLR tables. See the `lastAltered` field. Save that value somewhere for reference

Test:
- Re-run `DYNAMODB_URL="http://localhost:8000" S3_LOCAL_ENDPOINT="http://localhost:4569" node services/database/scripts/sync-kafka-2024.js`
- See in the console that the output says that the mcpar and mlr tables and buckets each had 1 item touched
  - In your services/uploads/local_buckets folder, verify the objects in the `fieldData` folder for MCPAR and MLR buckets have a new file that ends with `_S3rver_tagging.xml`. Open it to see the tag added to the bucket items: `lastSyncedByTag`
- At localhost:8001 view your MCPAR and MLR tables. Compare the new value of `lastAltered` with the ones you saved earlier

### Notes




---
### Pre-review checklist
